grpc/src/client.rs

```diff
@@ -44,24 +44,45 @@ enum ConnectionType {
 
 pub struct GRPCCommunicationsClient {
     name: String,
-    server_address: Url,
+    server_address: String,
     connection_type: ConnectionType,
 }
 
+fn verify_address_format(server_address: &String) -> Result<(), CommunicationMiddlewareError> {
+    let re = Regex::new(r"^https?:\/\/.+").unwrap_or_illegal_state();
+    if !re.is_match(server_address) {
+        return Err(CommunicationMiddlewareError(format!(
```

Comment:
Could you double-check the function coverage, please? The coverage report says that this return value (error case) is not covered by any test.

Reply:
I don't see the benefit of adding a utest here. We don't have a complicated algorithm and the regex is fairly simple.
The overall coverage of the file is indeed under our target, but this is caused by the requirements of the functionality which demand itests only. 
If you don't mind, I'll propose to be pragmatic here and leave out the utest as it only would bring maintenance effort without any added benefit.